tests/algorithms/test_snpe_posterior.py

```diff
@@ -0,0 +1,38 @@
+import pytest
+import torch
+import random
+
+import sbibm
+from sbibm.algorithms.sbi.snpe import run as run_posterior
+from sbibm.metrics.c2st import c2st
+
+#a fast test
+@pytest.mark.parametrize(
+    "task_name,num_observation",
+    [
+        (task_name, num_observation)
+        for task_name in ["gaussian_linear", "slcp",]
+        for num_observation in random.sample(range(1, 11), 2)
+    ],
+)
+def test_npe_posterior(
+    task_name, num_observation, num_simulations=1_000, num_samples=100
+):
+    task = sbibm.get_task(task_name)
+
+    samples = run_posterior(
+        task=task,
+        num_observation=num_observation,
+        num_simulations=num_simulations,
+        num_samples=num_samples,
+        num_rounds=1,
+        neural_net="mdn" #fast test
+    )
+
+    reference_samples = task.get_reference_posterior_samples(
+        num_observation=num_observation
+    )
+
+    acc = c2st(samples, reference_samples[:num_samples, :])
+
+    assert torch.abs(acc - 0.5) < 0.025
```

Comment:
I assume that we can't do SLCP with a MDN and 1000 simulations. What do want to target with this test -- correctness of API usage or correctness of the algorithm?

Reply:
Indeed, see [my comment earlier](https://github.com/sbi-benchmark/sbibm/pull/24#discussion_r748145215).

Reply:
see 9e91818